### PR TITLE
Int8 miopen call (gemm and conv)

### PR DESCRIPTION
**Pull request by scxiao**

### CODE REVIEW

File: src/targets/cpu/lowering.cpp [hunk 1]
```diff
@@ -204,6 +204,60 @@ struct cpu_convolution
...
+            auto in_h = in[2];
+            auto in_w = in[3];
+
+            auto wei   = weights.get_shape().lens();
+            auto wei_n = wei[0];
+            auto wei_c = wei[1];
+            auto wei_h = wei[2];
+            auto wei_w = wei[3];
+
+            par_dfor(output_shape.lens()[0],
```
Comment (kahmed10) [inline src/targets/cpu/lowering.cpp]: Can we replace this portion with a function shared with cpu_convolution? The only difference I see is the type of acc, other than that it looks the same.

File: src/targets/gpu/quant_convolution.cpp [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+                                        &beta,
+                                        x_desc_vec4.get(),
+                                        arg_vec4_x.implicit());
+    if(status != miopenStatusSuccess)
+    {
+        MIGRAPHX_THROW("QUANT_CONVOLUTION: transform input tensor failed");
+    }
+
+    // pack input to vec4 format
+    status = miopenTransformTensor(ctx.get_stream().get_miopen(),
```
Comment (pfultz2) [inline src/targets/gpu/quant_convolution.cpp]: This should be done in a separate operator that is inserted during lowering.

File: src/targets/gpu/quant_gemm.cpp [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+    auto dim_1      = n_dim - 1;
+    auto dim_0      = n_dim - 2;
+    auto arg_num    = args.size();
+    rocblas_int lda = args[0].get_shape().strides()[transa ? dim_1 : dim_0];
+    rocblas_int ldb = args[1].get_shape().strides()[transb ? dim_1 : dim_0];
+    rocblas_int ldc = args[arg_num - 1].get_shape().strides()[dim_0];
+
+    if(!transb)
+    {
+        device::pack_a(ctx.get_stream().get(), args[arg_num - 2], args[1]);
```
Comment (pfultz2) [inline src/targets/gpu/quant_gemm.cpp]: This should be done as a separate operator and inserted during lowering.

File: src/targets/gpu/quant_gemm.cpp [hunk 4]
```diff
@@ -0,0 +1,187 @@
...
+    }
+
+    // need to pack A in this scenario, use the algorithm to pack B in the
+    // comment of the API
+    if(transa)
+    {
+        device::pack_b(ctx.get_stream().get(), args[arg_num - 3], args[0]);
```
Comment (pfultz2) [inline src/targets/gpu/quant_gemm.cpp]: This should be done as a separate operator and inserted during lowering.

File: src/targets/cpu/lowering.cpp (hunk 1 above)
Comment (scxiao) [reply src/targets/cpu/lowering.cpp]: there is another difference that input and output types are different in quant_convolution.

File: src/targets/gpu/quant_gemm.cpp (hunk 3 above)
Comment (scxiao) [reply src/targets/gpu/quant_gemm.cpp]: done

File: src/targets/gpu/quant_convolution.cpp (hunk 2 above)
Comment (scxiao) [reply src/targets/gpu/quant_convolution.cpp]: fixed

File: src/targets/gpu/quant_gemm.cpp (hunk 4 above)
Comment (scxiao) [reply src/targets/gpu/quant_gemm.cpp]: fixed

**Pull request merged**